src/lib/import-export/tests/export/exporters/default-exporter.test.ts

```diff
@@ -0,0 +1,201 @@
+		( fsPromises.unlink as jest.Mock ).mockResolvedValue( undefined );
+		( fsPromises.mkdtemp as jest.Mock ).mockResolvedValue( '/tmp/studio_export_123' );
+		( fsPromises.writeFile as jest.Mock ).mockResolvedValue( undefined );
+		( os.tmpdir as jest.Mock ).mockReturnValue( '/tmp' );
+
+		mockArchiver.finalize.mockImplementation( () => {
+			return new Promise< void >( ( resolve ) => {
+				// Simulate async finalize
+				setTimeout( () => {
+					mockWriteStream.on.mock.calls.find( ( call ) => call[ 0 ] === 'close' )[ 1 ]();
+					resolve();
+				}, 0 );
+			} );
+		} );
+		exporter = new DefaultExporter( mockBackup );
+	} );
+
+	it( 'should create a tar.gz archive', async () => {
+		await exporter.export( mockOptions );
+
+		expect( archiver ).toHaveBeenCalledWith( 'tar', { gzip: true, gzipOptions: { level: 9 } } );
+	} );
+
+	it( 'should create a zip archive when the backup file ends with .zip', async () => {
+		mockOptions.backupFile = '/path/to/backup.zip';
+		await exporter.export( mockOptions );
+
+		expect( archiver ).toHaveBeenCalledWith( 'zip', { gzip: false, gzipOptions: undefined } );
+	} );
+
+	it( 'should add wp-config.php to the archive', async () => {
+		const options = {
+			...mockOptions,
+			includes: {
+				uploads: false,
+				plugins: false,
+				themes: false,
+				database: false,
+			},
+		};
+		await exporter.export( options );
+
+		expect( mockArchiver.file ).toHaveBeenCalledWith( '/path/to/wp-config.php', {
+			name: 'wp-config.php',
+		} );
+	} );
+
+	it( 'should add wp-content files to the archive', async () => {
+		const options = {
+			...mockOptions,
+			includes: {
+				uploads: true,
+				plugins: true,
+				themes: true,
+				database: false,
+			},
+		};
+		await exporter.export( options );
+
+		expect( mockArchiver.file ).toHaveBeenNthCalledWith( 1, '/path/to/wp-config.php', {
+			name: 'wp-config.php',
+		} );
+		expect( mockArchiver.file ).toHaveBeenNthCalledWith(
+			2,
+			'/path/to/wp-content/uploads/file1.jpg',
+			{
+				name: '../wp-content/uploads/file1.jpg',
+			}
+		);
+		expect( mockArchiver.file ).toHaveBeenNthCalledWith( 3, '/path/to/wp-content/plugins/plugin1', {
+			name: '../wp-content/plugins/plugin1',
+		} );
+		expect( mockArchiver.file ).toHaveBeenNthCalledWith( 4, '/path/to/wp-content/themes/theme1', {
+			name: '../wp-content/themes/theme1',
+		} );
+	} );
+
+	it( 'should add a database file to the archive when database is included', async () => {
+		const options = {
+			...mockOptions,
+			includes: {
+				plugins: false,
+				uploads: false,
+				themes: false,
+				database: true,
+			},
+		};
+		( fsPromises.mkdtemp as jest.Mock ).mockResolvedValue( '/tmp/studio_export_123' );
+
+		await exporter.export( options );
+
+		expect( mockArchiver.file ).toHaveBeenNthCalledWith( 1, '/path/to/wp-config.php', {
+			name: 'wp-config.php',
+		} );
+		expect( fsPromises.writeFile ).toHaveBeenCalledWith(
+			'/tmp/studio_export_123/file.sql',
+			'--test'
+		);
```

Comment:
I presume this shouldn't be included in the test as it was included to mock up the functionality.